src/niri.rs

```diff
@@ -891,20 +911,26 @@ impl State {
             );
 
             // Tell the windows their new focus state for window rule purposes.
-            if let KeyboardFocus::Layout {
+            let previous_focus_id = if let KeyboardFocus::Layout {
```

Comment:
I think it'll be cleaner to have `let mut previous_focus_id = None;` and then set it to `Some(...)` inside the condition.